storage/src/test/scala/io/delta/storage/integration/S3LogStoreUtilIntegrationTest.scala

```diff
@@ -0,0 +1,108 @@
+package io.delta.storage.integration
+
+import io.delta.storage.internal.{FileNameUtils, S3LogStoreUtil}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.fs.s3a.{S3AFileSystem, UploadInfo}
+import org.scalatest.Tag
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalactic.source
+
+import java.net.URI
+import java.nio.file.Paths
+import scala.math.max
+import scala.math.ceil
+import scala.math.round
+
+class S3LogStoreUtilIntegrationTest extends AnyFunSuite {
+  private val runIntegrationTests: Boolean =
+    Option(System.getenv("S3_LOG_STORE_UTIL_TEST_ENABLED")).exists(_.toBoolean)
+  private val bucket = System.getenv("S3_LOG_STORE_UTIL_TEST_BUCKET")
+  private val testRunUID =
+    System.getenv("S3_LOG_STORE_UTIL_TEST_RUN_UID") // Prefix for all S3 keys in the current run
+  private lazy val fs: S3AFileSystem = {
+    val fs = new S3AFileSystem()
+    fs.initialize(new URI(s"s3a://$bucket"), configuration)
+    fs
+  }
+  private val configuration = new Configuration()
+  configuration.set( // for local testing only
+    "fs.s3a.aws.credentials.provider",
+    "com.amazonaws.auth.profile.ProfileCredentialsProvider"
+  )
+
+  private val file = Paths.get(getClass.getClassLoader.getResource("sample.json").getPath).toFile
+
+  private def touch(key: String): UploadInfo =
+    fs.putObject(fs.newPutObjectRequest(key, fs.newObjectMetadata(), file))
```

Comment:
Can we use file system APIs to write the file? Then we don't need to add a sample.json file.

e.g.,
```
private lazy val fs = new Path(s"s3a://$bucket").getFileSystem(configuration)

private def touch(path: String): Unit = {
  fs.create(new Path(path)).close()
}
```